mlrun/runtimes/sparkjob/abstract.py

```diff
@@ -238,11 +238,17 @@ def deploy_default_image(cls, with_gpu=False):
         get_run_db().delete_function(name=sj.metadata.name)
 
     def _is_using_gpu(self):
-        _, driver_gpu = self._get_gpu_type_and_quantity(
-            resources=self.spec.driver_resources["limits"]
+        driver_limits = self.spec.driver_resources.get("limits")
```

Comment:
```suggestion
        driver_limits = self.spec.driver_resources.get("limits")
        driver_gpu = None
        if driver_limits:
            _, driver_gpu = self._get_gpu_type_and_quantity(resources=driver_limits)

        executor_limits = self.spec.executor_resources.get("limits")
        executor_gpu = None
        if executor_limits:
            _, executor_gpu = self._get_gpu_type_and_quantity(resources=executor_limits)
```
This might be a tad simpler.